### PR TITLE
Change to src/ layout

### PR DESCRIPTION
**Pull request by nicoddemus**
This is nowdays considered best practice, and in line with our other repos on GitHub.

### CODE REVIEW

**Review by oliveira-mauricio — APPROVED** [review]

File: base_environment.yml [hunk 1]
```diff
@@ -12,3 +12,7 @@ dependencies:
     - pip
     - pre-commit
     - pyqt =5.12
+
+environment:
```
Comment (prusse-martin) [inline base_environment.yml:16]: This is not a `.devenv.yml` file there is no job failing, but this file is used to generate the docs.
This should not be a problem, I think conda ignores this.

**Review by prusse-martin — APPROVED** [review]

File: base_environment.yml (hunk 1 above)
Comment (prusse-martin) [reply base_environment.yml:16]: I just check devenv.
if we throw a file whitout `.devenv.yml` extension there is no include or environment handling. If some one includes this file they are handled.

**Comment by prusse-martin** [comment]
You will probably also want to update https://github.com/ESSS/qmxgraph/blob/master/docs/conf.py#L23.
But since the documentation build is failing we can postpone this.

**Comment by prusse-martin** [comment]
I did add 2 items to #37:
- Fix documentation build;
- Check and update examples;

So we can merge this and review the docs later;

File: base_environment.yml (hunk 1 above)
Comment (nicoddemus) [reply base_environment.yml:16]: Good catch.

I changed it to `devenv.yml` for consistency and renamed to `_base_environment.devenv.yml` to signal its internal status. 👍

**Comment by nicoddemus** [comment]
> You will probably also want to update https://github.com/ESSS/qmxgraph/blob/master/docs/conf.py#L23.
But since the documentation build is failing we can postpone this.

Good point, but went ahead and removed this: readthedocs has the option to install the project into a virtualenv to build the docs, which removes the need for that workaround. But we still need to fix #37 to get the documentation up and running, as you mentioned.

**Pull request merged**